components/tensorflow-notebook-image/start-notebook.sh

```diff
@@ -15,9 +15,5 @@
 
 set -e
 
-if [[ ! -z "${JUPYTERHUB_API_TOKEN}" ]]; then
-  # launched by JupyterHub, use single-user entrypoint
-  exec /usr/local/bin/start-singleuser.sh $*
-else
-  . /usr/local/bin/start.sh jupyter notebook $*
-fi
+jupyter notebook --notebook-dir=/home/jovyan --ip=0.0.0.0 --no-browser --allow-root --port=8888
```

Comment:
Why not just remove this file and put this line in Dockerfile?
https://github.com/kubeflow/kubeflow/blob/master/components/tensorflow-notebook-image/Dockerfile#L157

Reply:
start-notebook.sh is being referenced in Dockerfile of kubeflow/componnets/contrib images. So I reverted changes for this file.

Reply:
@lluunn , @zabbasi This is a great fix, because it exposes the full arguments to Jupyter in the entry point, without changing the actual contents of the image, if I understand correctly.

So building one more image with a new entrypoint, should be a very light operation, which does not require creating new layers. Maybe we can have our cake and eat it this way:
* All images are self-standing, they just start with default settings if I run `docker run`
* Since TFX widgets do not work right now with JupyterLab, Kubeflow can propose to users that they use images that start the old interface.
* Users can just choose another image in the web app from the list, and switch to a new image, which has the same layers, but a different entrypoint, to start the current JupyterLab interface.

WDYT @jlewi ?

Reply:
@vkoukis See #2208. If we're building jupyter images I think the question we should ask is why are we building Kubeflow specific Jupyter images and then based on that figure out the right thing to do.

For example the TensorFlow docker image comes with Jupyter and I believe starts it automatically. So why can't users just use that why does Kubeflow need to build a new Jupyter image?

Once we pick a reason such as "Jupyter Environment suitable for TFX" or Jupyter environment for PyTorch or NVIDIA Rapids, i think it will be easier to figure out which libraries should be installed in that notebook and what environment should be used.

Historically, the reason we got into the business of building Jupyter images is because we were using JupyterHub and that required customization of the images.

Reply:
> For example the TensorFlow docker image comes with Jupyter and I believe starts it automatically. So why can't users just use that why does Kubeflow need to build a new Jupyter image?

I'm not sure. I think the biggest blocker is that these generic images will not contain Kubeflow-specific components, like Fairing, which are included in images built by Kubeflow. I'll look more into it though.

> Once we pick a reason such as "Jupyter Environment suitable for TFX" or Jupyter environment for PyTorch or NVIDIA Rapids, i think it will be easier to figure out which libraries should be installed in that notebook and what environment should be used.

Agreed. I understand that currently you opt for Kubeflow to use the older Jupyter interface, so you maintain compatibility with TFX widgets that misbehave with JupyterLab [Are you referring to [this](https://github.com/kubeflow/kubeflow/issues/1130)? Maybe this will be fixed eventually in the widget code? ]
This is a certainly a strong argument.

I would also argue that it's important for Kubeflow to showcase the latest JupyterLab UI along with its components. I'll make a note in the next community meeting to solicit feedback, and we'd be happy to make a PR to Kubeflow so we can build and maintain JupyterLab-enabled images that also contain its custom components.